File: daily/2025/2561_Rearranging_Fruits.cpp
#include <iostream>
#include <vector>

using namespace std;

class Solution {
public:
    long long minCost(vector<int>& basket1, vector<int>& basket2) {
        unordered_map<int,int> cnt;
        for (int v : basket1) cnt[v]++;
        for (int v : basket2) cnt[v]--;
        vector<int> m;
        int cost = INT_MAX;
        for (auto& [v, c] : cnt){
            if (c % 2) return -1;
            cost = min(cost, v);
            for (int i = 0; i < abs(c) / 2; i++) m.push_back(v);
        }
        sort(m.begin(), m.end());
        long long ans = 0;
        for (int i = 0; i < m.size() / 2; i++){
            ans += min(cost * 2, m[i]);
        }
        return ans;
    }
};

int main(){
    vector<int> basket1 = {4,2,2,2};
    vector<int> basket2 = {1,4,1,2};
    // vector<int> basket1 = {2,3,4,1};
    // vector<int> basket2 = {3,2,5,1};
    // vector<int> basket1 = {84,80,43,8,80,88,43,14,100,88};
    // vector<int> basket2 = {32,32,42,68,68,100,42,84,14,8};
    Solution S;
    cout << S.minCost(basket1, basket2) << endl;
    return 0;
}

File: daily_python/2561_Rearranging_Fruits.py
import os
import sys
from typing import List
from collections import defaultdict

class Solution:
    def minCost(self, basket1: List[int], basket2: List[int]) -> int:
        cnt = defaultdict(int)
        for f in basket1:
            cnt[f] += 1
        for f in basket2:
            cnt[f] -= 1
        move = []
        m = float('inf')
        for v, c in cnt.items():
            if c % 2 == 1:
                return -1
            m = min(m, v)
            for i in range(abs(c) // 2):
                move.append(v)
        move.sort()
        ans = 0
        for i in range(len(move) // 2):
            ans += min(m * 2, move[i])
        print(move)
        return ans
    
def readDataSet(filename):
    dataset = []
    with open(filename, 'r') as file:
        content = file.read().strip()
        blocks = content.split('\n\n')
        for block in blocks:
            lines = block.split('\n')
            basket1 = list(map(int, lines[0].split('=')[1].strip()[1:-2].split(',')))
            basket2 = list(map(int, lines[1].split('=')[1].strip()[1:-2].split(',')))
            dataset.append((basket1, basket2))
    return dataset

if __name__ == '__main__':
    if len(sys.argv) == 1:
        filename = os.path.splitext(__file__)[0] + '.txt'
    else:
        filename = sys.argv[1]
    dataset = readDataSet(filename)
    results = []
    solution = Solution()
    for basket1, basket2 in dataset:
        results.append(solution.minCost(basket1, basket2))
    for index, result in enumerate(results):
        print(f'Example {index + 1} : {result}')